### PR TITLE
feat: benchmarking issue template

### LINKED ISSUE #382: [FEATURE] Internal benchmarking issue template for dev team
**Please describe the purpose of the feature. Is it related to a problem?**
A template focused on internal benchmarking tasks.

**Describe the solution you'd like**
A benchmarking issue template focused on planned benchmarking tasks from the dev team.

**How do we know when implementation of this feature is complete?**
Checklist:

- [x] Issue template md file.

### PR DESCRIPTION
**Pull request by arnupretorius**
## What?
Issue template for benchmarking tasks.
## Why?
To organise/plan benchmarking work.
## How?
Add markdown template.

closes #382

### CODE REVIEW

**Review by RuanJohn — COMMENTED** [review]
I think this looks good. Is the general idea here that we would always open an issue on github before we run benchmarks?

**Comment by arnupretorius** [comment]
> I think this looks good. Is the general idea here that we would always open an issue on github before we run benchmarks?

Yes, that would be the idea. I would also like us to have benchmark results be visible on the repo itself, so in the future benchmarking will move closer to Mava as opposed to the results only being available on the experiment runner.

**Comment by RuanJohn** [comment]
> > I think this looks good. Is the general idea here that we would always open an issue on github before we run benchmarks?
> 
> Yes, that would be the idea. I would also like us to have benchmark results be visible on the repo itself, so in the future benchmarking will move closer to Mava as opposed to the results only being available on the experiment runner.

Sounds good.

**Review by RuanJohn — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/benchmarking_internal.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+A clear and concise description of what you expected to happen.
+
+### Definition of done
+A precise outline for the experiment or task to be considered complete.
+
+### Mandatory checklist before making a PR
+* [ ] The success criteria laid down in “Definition of done” are met.
+* [ ] Experiment is documented - hyperparameters, plots, conclusions/findings etc.
+* [ ] Documentation is updated - README, CONTRIBUTING, or other documentation.
+* [ ] Link experiment/benchmarking.
```
Comment (KaleabTessera) [inline .github/ISSUE_TEMPLATE/benchmarking_internal.md:23]: This last point should maybe be optional. I am not sure if we can make experiment logs public according to our current setup.

**Review by KaleabTessera — APPROVED** [review]
Looks good. Just a minor comment.

File: .github/ISSUE_TEMPLATE/benchmarking_internal.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+### Describe the benchmarking experiment/task
+A clear and concise description of what the task is. If it is an experiment please provide the experimental design.
+
+### Hypothesis/expected behavior or outcome
+A clear and concise description of what you expected to happen.
+
+### Definition of done
+A precise outline for the experiment or task to be considered complete.
+
+### Mandatory checklist before making a PR
```
Comment (KaleabTessera) [inline .github/ISSUE_TEMPLATE/benchmarking_internal.md:19]: This should say probably read "Mandatory checklist before benchmarking is complete". Benchmarks are currently on run on PRs.

File: .github/ISSUE_TEMPLATE/benchmarking_internal.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+* [ ] The success criteria laid down in “Definition of done” are met.
```
Comment (KaleabTessera) [inline .github/ISSUE_TEMPLATE/benchmarking_internal.md:20]: We should also include this step -  `* [ ] PR for code to benchmark has been created and the label has been changed to `benchmark` `

File: .github/ISSUE_TEMPLATE/benchmarking_internal.md (hunk 2 above)
Comment (arnupretorius) [reply .github/ISSUE_TEMPLATE/benchmarking_internal.md]: Done.

File: .github/ISSUE_TEMPLATE/benchmarking_internal.md (hunk 3 above)
Comment (arnupretorius) [reply .github/ISSUE_TEMPLATE/benchmarking_internal.md:21]: Added this

File: .github/ISSUE_TEMPLATE/benchmarking_internal.md (hunk 1 above)
Comment (arnupretorius) [reply .github/ISSUE_TEMPLATE/benchmarking_internal.md]: Done

**Review by KaleabTessera — APPROVED** [review]
Thanks @arnupretorius !

**Review by RuanJohn — APPROVED** [review]
Everything looks good to me. Thank you @arnupretorius :)

**Pull request merged**